tests/ResourceTests.cpp

```diff
@@ -276,6 +276,36 @@ void UniqueProcessInfo()
 }
 #endif
 
+// Compilation only test...
+#ifdef WIL_ENABLE_EXCEPTIONS
+void NoexceptConstructibleTest()
+{
+    using BaseStorage = wil::details::unique_storage<wil::details::handle_resource_policy>;
+
+    struct ThrowingConstructor : BaseStorage
+    {
+        explicit ThrowingConstructor(HANDLE) __WI_NOEXCEPT_(false) {}
+    };
+
+    struct ProtectedConstructor : BaseStorage
+    {
+    protected:
+        explicit ProtectedConstructor(HANDLE) WI_NOEXCEPT {}
+    };
+
+    // wil::unique_handle is one of the many types which are expected to be noexcept
+    // constructible since they don't perform any "advanced" initialization.
+    static_assert(wistd::is_nothrow_constructible_v<wil::unique_handle, HANDLE>, "wil::unique_any_t should be noexcept if the storage is");
```

Comment:
Maybe also assert that it's nothrow constructable with no parameters? Just for completeness.